packages/js/src/plugins/auctionHouseModule/operations/executeSale.ts

```diff
@@ -324,12 +324,16 @@ export const executeSaleBuilder = (
     throw new AuctioneerPartialSaleNotSupportedError();
   }
   if (isPartialSale) {
+    const listingPricePerToken = price.basisPoints.div(tokens.basisPoints);
     const buyerPricePerToken = buyerPrice.basisPoints.div(
       buyerTokensSize.basisPoints
     );
 
-    if (!price.basisPoints.eq(buyerPricePerToken.mul(tokens.basisPoints))) {
-      throw new PartialPriceMismatchError();
+    if (!listingPricePerToken.eq(buyerPricePerToken)) {
+      throw new PartialPriceMismatchError({
+        expected: lamports(listingPricePerToken),
+        actual: lamports(buyerPricePerToken),
```

Comment:
This will not be lamports if the Auction House does not use SOL as a currency. I believe we have that information on the `auctionHouse` model.